app/lib/shared/user_merger.dart

```diff
@@ -0,0 +1,191 @@
+      (Package m) async {
+        await _db.withTransaction((tx) async {
+          final p = (await tx.lookup<Package>([m.key])).single;
+          if (p.containsUploader(fromUserId)) {
+            p.removeUploader(fromUserId);
+            p.addUploader(toUserId);
+            tx.queueMutations(inserts: [p]);
+            await tx.commit();
+          } else {
+            await tx.rollback();
+          }
+        });
+      },
+    );
+
+    // PackageVersion
+    await _processConcurrently(
+      _db.query<PackageVersion>()..filter('uploader =', fromUserId),
+      (PackageVersion m) async {
+        await _db.withTransaction((tx) async {
+          final pv = (await tx.lookup<PackageVersion>([m.key])).single;
+          if (pv.uploader == fromUserId) {
+            pv.uploader = toUserId;
+            tx.queueMutations(inserts: [pv]);
+            await tx.commit();
+          } else {
+            await tx.rollback();
+          }
+        });
+      },
+    );
+
+    // UserSession
+    final fromUserKey = _db.emptyKey.append(User, id: fromUserId);
+    final toUserKey = _db.emptyKey.append(User, id: toUserId);
+    await _processConcurrently(
+      _db.query<UserSession>()..filter('userIdKey =', fromUserKey),
+      (UserSession m) async {
+        await _db.withTransaction((tx) async {
+          final session = (await tx.lookup<UserSession>([m.key])).single;
+          if (session.userId == fromUserId) {
+            session.userIdKey = toUserKey;
+            tx.queueMutations(inserts: [session]);
+            await tx.commit();
+          } else {
+            await tx.rollback();
+          }
+        });
+      },
+    );
+
+    // Consent of the User
+    await _processConcurrently(
+      _db.query<Consent>(ancestorKey: fromUserKey),
+      (Consent m) async {
+        await _db.withTransaction((tx) async {
+          tx.queueMutations(
+              inserts: [m.changeParentUserId(toUserId)], deletes: [m.key]);
+          await tx.commit();
+        });
+      },
+    );
+
+    // Consent's fromUserId attribute
+    await _processConcurrently(
+      _db.query<Consent>()..filter('fromUserId =', fromUserId),
+      (Consent m) async {
+        await _db.withTransaction((tx) async {
+          final consent = (await tx.lookup<Consent>([m.key])).single;
+          if (consent.fromUserId == fromUserId) {
+            consent.fromUserId = toUserId;
+            tx.queueMutations(inserts: [consent]);
+            await tx.commit();
+          } else {
+            await tx.rollback();
+          }
+        });
+      },
+    );
+
+    // PublisherMember
+    await _processConcurrently(
+      _db.query<PublisherMember>()..filter('userId =', fromUserId),
+      (PublisherMember m) async {
+        await _db.withTransaction((tx) async {
+          final pm = (await tx.lookup<PublisherMember>([m.key])).single;
+          if (pm.userId == fromUserId) {
+            pm.userId = toUserId;
```

Comment:
This won't work, the entity also has `id = userId`, thus, we have to delete and create the entity.